pkg/registry/generic/rest/streamer.go

```diff
@@ -56,6 +59,16 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
 	if err != nil {
 		return
 	}
+
+	if resp.StatusCode == http.StatusInternalServerError {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
This object is to allow someone to return a stream as a response instead of
an object.  The behavior of this streamer should be that if the streamer
wants to return an error, it should do so via err.

On Mon, Sep 14, 2015 at 12:40 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/registry/generic/rest/streamer.go
> <https://github.com/kubernetes/kubernetes/pull/13722#discussion_r39416075>
> :
>
> > @@ -56,6 +59,16 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
> >  	if err != nil {
> >  		return
> >  	}
> > +
> > +	if resp.StatusCode == http.StatusInternalServerError {
> > +		bodyBytes, err := ioutil.ReadAll(resp.Body)
>
> Oh, I did not realize that. *reads code* In that case I think it's a big
> mistake to do any interpretation here at all, the stream should be passed
> to clients exactly as its received.
>
> ... why do we even have this object at all? It should just be a wrapper
> around the rest of our proxying code.
>
> @feihujiang <https://github.com/feihujiang> I guess your goal here was to
> get the HTTP status code repeated back to clients properly? I think you're
> going to have to fix the interface & callers, there's no way to do that
> properly with this interface.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13722/files#r39416075>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift